server/channels/app/product_notices.go

```diff
@@ -94,10 +76,9 @@ func noticeMatchesConditions(config *model.Config, preferences store.PreferenceS
 
 	// check if current server version is notice range
 	if !isCloud && cnd.ServerVersion != nil {
-		version := cleanupVersion(model.BuildNumber)
-		serverVersion, err := semver.NewVersion(version)
+		serverVersion, err := semver.NewVersion(model.CurrentVersion)
```

Comment:
@toninis Would this current version have any of the prefix or suffix that the cleanupVersion function was doing above? I believe the intent of that was it does a direct comparison with the versions that are included in the [notices.json](https://github.com/mattermost/notices/blob/master/notices.json) so it needs to clean up any additional prefix so it matches with the version strings in that notices definition

Reply:
Current version should get the last value from https://github.com/mattermost/mattermost/blob/master/server/public/model/version.go#L15 always

Reply:
Change sounds good. However, the tests seem to be failing in cases where the version has additional info like hashes and this `cleanupVersion` would remove it. If we no longer support those cases then we need to remove the tests.

Reply:
Yep . I need to fix the test 😄